### PR TITLE
[App Service] Fix App Service config appsettings set and delete to redact secret values

### PR DESCRIPTION
**Pull request by amberwang113**
**Related command**
az webapp config appsettings set/delete
az logicapp config appsettings set/delete 
az functionapp config appsettings set/delete

**Description**

Sets default output for the above commands to hide values. Guards against accidental security leaks through CI logs where non-authorized users can see returned secrets in appsettings. Adds parameter --show-values; when present, values will be displayed as before.

**Testing Guide**

azdev test test_webapp_config_appsettings -> addition of the --show-values parameter results in an output with all appsettings shown, whereas excluding it results in the values being redacted

**History Notes**

[App Service] `az webapp config appsettings set/delete`: Update App Service config appsettings set and delete to redact secret values
[App Service] `az logicapp config appsettings set/delete`: Update App Service config appsettings set and delete to redact secret values
[App Service] `az functionapp config appsettings set/delete`: Update App Service config appsettings set and delete to redact secret values

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -1597,11 +1598,12 @@ def _ssl_context():
...
 
 
-def _build_app_settings_output(app_settings, slot_cfg_names):
+def _build_app_settings_output(app_settings, slot_cfg_names, show_values=True):
     slot_cfg_names = slot_cfg_names or []
+    app_settings = _mask_creds_related_appsettings(app_settings)
     return [{'name': p,
              'value': app_settings[p],
-             'slotSetting': p in slot_cfg_names} for p in _mask_creds_related_appsettings(app_settings)]
+             'slotSetting': p in slot_cfg_names} for p in (app_settings if show_values else _redact_appsettings(app_settings))]
```
Comment (naveedaz) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:1606]: We should maintain the old behavior if show_values is set to true. In this case, we should still call _mask_creds_related_appsettings

Comment (amberwang113) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:1606]: I added line 1603 appsettings = _mask_creds_related_appsettings, which should overshadow (and hopefully make it easier to delete that once it gets fixed and is no longer necessary)

**Comment by zhoxing-ms** [comment]
@amberwang113 Could you please resolve these CI style issues?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:1606]: `_mask_creds_related_appsettings` only masks the value of `DOCKER_REGISTRY_SERVER_PASSWORD`:

https://github.com/Azure/azure-cli/blob/3024c5c86f22f9a0f6cc7def626d458d06d62267/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L1808-L1811

https://github.com/Azure/azure-cli/blob/3024c5c86f22f9a0f6cc7def626d458d06d62267/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L1696

Masking `DOCKER_REGISTRY_SERVER_PASSWORD` may not be necessary if the user explicitly asks to show values.

**Review by jiasli — CHANGES_REQUESTED** [review]
Please reconsider the design.

**Comment by kamperiadis** [comment]
Based on our chat with everyone yesterday, we might want to just return "Operation succeeded" or "Settings X, Y, Z are successfully updated" since redacting would not really communicate to customers that the app settings were successfully updated when running `config appsettings set`.

**Comment by amberwang113** [comment]
closed because moving forward with https://github.com/Azure/azure-cli/pull/27565 option to output nothing over redacted values

**Pull request closed without merging**